pulp_container/app/registry_api.py

```diff
@@ -449,6 +452,19 @@ class CatalogView(ContainerRegistryApiMixin, ListAPIView):
     queryset = models.ContainerDistribution.objects.all().only("base_path")
     serializer_class = ContainerCatalogSerializer
     pagination_class = ContainerCatalogPagination
+    access_policy_class = RegistryAccessPolicy()
+
+    def get_queryset(self, *args, **kwargs):
+        """Filter the queryset based on assigned permissions."""
+        queryset = super().get_queryset()
+
+        accessible_repo_pks = []
+        for obj in queryset:
```

Comment:
This is looping over the complete queryset. We should not do that, but extend the queryset with rules that can be translated into SQL.

Reply:
Discussion of the container meeting:
We should use `get_objects_for_user` and filter for `public=True`.
This will hard code the queryset, but that is at least in line with what we do elsewhere. But it will not digest the whole db-table for every single call.